Tasks/Gradle/sonarqube-common/sonarqube-server.ts

```diff
@@ -0,0 +1,112 @@
+import Q = require('q');
+import url = require('url');
+import https = require('https');
+import http = require('http');
+import {IncomingMessage} from 'http';
+
+import {SonarQubeEndpoint} from './sonarqube-common';
+import sqCommon = require('./sonarqube-common');
+
+import tl = require('vsts-task-lib/task');
+
+export interface ISonarQubeServer {
+    callSonarQubeRestEndpoint(path:string):Q.Promise<Object>;
+
+    getEndpoint():SonarQubeEndpoint;
+}
+
+export class SonarQubeServer implements ISonarQubeServer {
+
+    private endpoint:SonarQubeEndpoint;
+
+    constructor() {
+        this.endpoint = sqCommon.getSonarQubeEndpoint();
```

Comment:
Also, the server should be given the endpoint, it shouldn't have to know how to get it.